# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentChild.cpp
# Commit: 365223fc9f86
# Full Hash: 365223fc9f865212c0f42948763ec48d4dca03ff
# Author: Barret Rennie <barret@brennie.ca>
# Date: 2019-04-06 10:57:50
# Description:
#   Bug 1542183 - Ensure TabChilds always have a TabGroup upon initialization r=nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D26374
# ==============================================================================

diff -r 21358d3572a7 -r 365223fc9f86 dom/ipc/ContentChild.cpp
--- a/dom/ipc/ContentChild.cpp	Fri Apr 05 19:25:30 2019 +0000
+++ b/dom/ipc/ContentChild.cpp	Fri Apr 05 19:38:27 2019 +0000
@@ -1819,6 +1819,14 @@
   }
 
   auto tabChild = static_cast<TabChild*>(aActor);
+  if (!tabChild->mTabGroup) {
+    tabChild->mTabGroup = TabGroup::GetFromActor(tabChild);
+
+    if (!tabChild->mTabGroup) {
+      tabChild->mTabGroup = new TabGroup();
+      MOZ_DIAGNOSTIC_ASSERT(aSameTabGroupAs != 0);
+    }
+  }
 
   if (NS_WARN_IF(NS_FAILED(tabChild->Init(/* aOpener */ nullptr)))) {
     return IPC_FAIL(tabChild, "TabChild::Init failed");